Mathlib/Logic/Equiv/Fin.lean

```diff
@@ -288,32 +288,36 @@ theorem finSuccEquivLast_symm_some (i : Fin n) :
 
 /-- Equivalence between `Π j : Fin (n + 1), α j` and `α i × Π j : Fin n, α (Fin.succAbove i j)`. -/
 @[simps (config := .asFn)]
-def Equiv.piFinSuccAboveEquiv (α : Fin (n + 1) → Type u) (i : Fin (n + 1)) :
+def Equiv.piFinSuccAbove (α : Fin (n + 1) → Type u) (i : Fin (n + 1)) :
     (∀ j, α j) ≃ α i × ∀ j, α (i.succAbove j) where
   toFun f := (f i, fun j => f (i.succAbove j))
   invFun f := i.insertNth f.1 f.2
   left_inv f := by simp [Fin.insertNth_eq_iff]
   right_inv f := by simp
-#align equiv.pi_fin_succ_above_equiv Equiv.piFinSuccAboveEquiv
-#align equiv.pi_fin_succ_above_equiv_apply Equiv.piFinSuccAboveEquiv_apply
-#align equiv.pi_fin_succ_above_equiv_symm_apply Equiv.piFinSuccAboveEquiv_symm_apply
+#align equiv.pi_fin_succ_above_equiv Equiv.piFinSuccAbove
+#align equiv.pi_fin_succ_above_equiv_apply Equiv.piFinSuccAbove_apply
+#align equiv.pi_fin_succ_above_equiv_symm_apply Equiv.piFinSuccAbove_symm_apply
 
 /-- Order isomorphism between `Π j : Fin (n + 1), α j` and
 `α i × Π j : Fin n, α (Fin.succAbove i j)`. -/
 def OrderIso.piFinSuccAboveIso (α : Fin (n + 1) → Type u) [∀ i, LE (α i)]
     (i : Fin (n + 1)) : (∀ j, α j) ≃o α i × ∀ j, α (i.succAbove j) where
-  toEquiv := Equiv.piFinSuccAboveEquiv α i
+  toEquiv := Equiv.piFinSuccAbove α i
   map_rel_iff' := Iff.symm i.forall_iff_succAbove
 #align order_iso.pi_fin_succ_above_iso OrderIso.piFinSuccAboveIso
 
 /-- Equivalence between `Fin (n + 1) → β` and `β × (Fin n → β)`. -/
 @[simps! (config := .asFn)]
 def Equiv.piFinSucc (n : ℕ) (β : Type u) : (Fin (n + 1) → β) ≃ β × (Fin n → β) :=
-  Equiv.piFinSuccAboveEquiv (fun _ => β) 0
+  Equiv.piFinSuccAbove (fun _ => β) 0
 #align equiv.pi_fin_succ Equiv.piFinSucc
 #align equiv.pi_fin_succ_apply Equiv.piFinSucc_apply
 #align equiv.pi_fin_succ_symm_apply Equiv.piFinSucc_symm_apply
 
+@[simps! (config := .asFn)]
```

Comment:
Does this actually generate a lemma in terms of `Fin.castSucc`? If not, I'd prefer to see the lemma written manually.

Reply:
Ok I removed it. I don't know what that syntax does I just saw it on the thing I copy-pasted to make this. Is it related to the thing Yaël was saying about bundling?

Reply:
Oh, or perhaps the additional align statements I see immediately above?

Reply:
Ok I tried adding the lemmas that I saw as align statements using the snoc-y API, but if I leave it as is `#check` indicates the lemmas already do things that way.

```
#check Equiv.piFinCastSucc_apply
-- Equiv.piFinCastSucc_apply.{u} (n : ℕ) (β : Type u) :
--  ⇑(Equiv.piFinCastSucc n β) = fun f ↦ (f (Fin.last n), fun j ↦ f (Fin.castSucc j))
#check Equiv.piFinCastSucc_symm_apply
-- Equiv.piFinCastSucc_symm_apply.{u} (n : ℕ) (β : Type u) : ⇑(Equiv.piFinCastSucc n β).symm = fun f ↦ Fin.snoc f.2 f.1
```

Reply:
You can use `@[simps?]` to ask simps what lemmas it generated.

Reply:
Ok I'm marking this resolved, lmk If I made a mistake.